cityofaustin/atd-vz-data
# Database -- automate insertion into fatalities table

## Associated issues
Closes cityofaustin/atd-data-tech/issues/11337

## Testing
**URL to test:** 
You will test this using Frank's new local development steps to replicate the production db onto your local machine and then add my DB triggers.

**Steps to test:**
1. Checkout this branch locally
2. In the root of the branch follow instructions in the README to start up a local instance of the VZ stack
3. Make sure you un-comment [this](https://github.com/cityofaustin/atd-vz-data/blob/e09b9ad73c9c5e5880ad9757975b930ec01fbec9/atd-vze/.env#L2) line and comment out the other one, I also had to change the "https" to "http" or else I get an Network connection error when trying to load data in the VZE.
4. You will need to add the `is_deleted` column to the `fatalities` table. It is a boolean and set it to default = false.
5. Add the new fatalities [view](https://github.com/cityofaustin/atd-vz-data/blob/3c9c5dfdce272759a01fb059198d15d06908f059/atd-vzd/views/view_fatalities.sql) to your local DB. Theres a SQL editor in the hasura console if you wanna use that, or use DBeaver.
6. Open up your localhost:8080 instance of the hasura console. Go to the new `view_fatalities` view you created and manually add a relationship to the recommendations table like so:
![Screenshot from 2023-03-08 12-42-14](https://user-images.githubusercontent.com/67130963/223884308-b55ca458-a5b1-4678-9035-5f8b0baf1ab2.png)
7. Copy and paste the triggers and trigger functions from [this](https://github.com/cityofaustin/atd-vz-data/files/10926497/fatalities_db_automation.txt) file into the hasura SQL editor or DBeaver. This will add the triggers and functions into the DB and their respective tables.
8. Now go to your localhost:3000 instance of the VZE. Go to the Fatalities tab, you will see an empty table.
9. In your local graphql engine test adding fatal entries to the `atd_txdot_primaryperson` or `atd_txdot_person` tables where the `prsn_injry_sev_id = 4`. This will add a fatality for that person to the `fatalities` table. If the fatality also occurred within the Austin jurisdiction, the fatality will appear in the `view_fatality` and thus in the front end Fatalities table in VZE.
10. Test deleting the entry you made in the `primary_person` or `person` table, this should also delete the entry in the `fatalities` table.
11. Deleting the crash from the `atd_txdot_crashes` table will also delete the record in the `fatalities` table.
---
#### Ship list
